From: ptrcnull <git@ptrcnull.me>
Date: Mon, 11 Mar 2024 00:42:17 +0100
Subject: [PATCH] Fix runtime failures on 32-bit systems

famous last words:
> If that breaks 32bit we wont notice for now
> as I think all the CI coverage is on 64bit at this point

Co-authored-by: Tasos Sahanidis <tasos@tasossah.com>

diff --git a/src/core/connectionbackend.cpp b/src/core/connectionbackend.cpp
index ef86e49..86a124b 100644
--- a/src/core/connectionbackend.cpp
+++ b/src/core/connectionbackend.cpp
@@ -176,3 +176,3 @@ bool ConnectionBackend::sendCommand(int cmd, const QByteArray &data) const
     // see https://invent.kde.org/frameworks/kio/-/merge_requests/1141#note_606633
-    sprintf(buffer, "%6llx_%2x_", data.size(), cmd);
+    sprintf(buffer, "%6zx_%2x_", (size_t)data.size(), cmd);
     socket->write(buffer, HeaderSize);
